R/module_teal_data.R

```diff
@@ -94,8 +81,61 @@ ui_validate_reactive_teal_data <- function(id) {
 #' @rdname module_teal_data
 srv_validate_reactive_teal_data <- function(id, # nolint: object_length
                                             data,
+                                            input_data,
                                             modules = NULL,
                                             validate_shiny_silent_error = FALSE) {
+  UseMethod("srv_validate_reactive_teal_data", input_data)
+}
+
+#' @rdname module_teal_data
+srv_validate_reactive_teal_data.default <- function(id, # nolint: object_length
+                                                    data,
+                                                    input_data,
+                                                    modules = NULL,
+                                                    validate_shiny_silent_error = FALSE) {
+  stop("Unsupported")
+}
+
+#' @rdname module_teal_data
```

Comment:
I think it requires `@export` tag to register S3 method. Otherwise dispatch won't find the method. Besides, I think this S3 complicates the case. I suggest to seek a simplification and generalisation in `srv_teal/srv_init_data/srv_teal_data`

```suggestion
#' @rdname module_teal_data
#' @export
```